internal/engine/upper.go

```diff
@@ -92,6 +92,14 @@ func NewUpper(ctx context.Context, b BuildAndDeployer, k8s k8s.Client,
 	}
 }
 
+func (u Upper) LogActionLogger(ctx context.Context) logger.Logger {
```

Comment:
maybe call this `NewLogActionLogger` or `MakeLogActionLogger`, or else "log" sounds like the verb here (like you are logging an action logger)